### PR TITLE
4.2.1 release notes etc

### PR DESCRIPTION
**Pull request by neworderofjamie**
Updates to the documentation and release notes

### CODE REVIEW

File: doxygen/00_MainPage.dox [hunk 1]
```diff
@@ -18,7 +18,7 @@ GeNN is a software package to enable neuronal network simulations on
 NVIDIA GPUs by code generation. Models are defined in a simple C-style
 API and the code for running them on either GPU or CPU hardware is
 generated by GeNN. GeNN can also be used through external
-interfaces. Currently there are interfaces for \ref SpineML and for [Brian](http://brian2.readthedocs.io) via \ref brian2genn "Brian2GeNN".
+interfaces. Currently there are interfaces for \ref SpineML, a \ref PyGeNN and a [Brian](http://brian2.readthedocs.io) interface via \ref brian2genn "Brian2GeNN".
```
Comment (tnowotny) [inline doxygen/00_MainPage.dox:21]: SpineML, \ref PyGeNN and [Brian]... (no "a"s)

File: doxygen/06_Brian2GeNN.dox [hunk 2]
```diff
@@ -1,9 +1,8 @@
...
-When Brian2GeNN is installed in this way, it comes with a bundled version of GeNN and no further configuration is required. In all other cases (e.g. an installation from source), the path to GeNN and the CUDA libraries has to be configured via the `GENN_PATH` and `CUDA_PATH` environment variables as described in \ref Installation or via the `devices.genn.path` and `devices.genn.cuda_path` [Brian preferences](http://brian2.readthedocs.io/en/stable/advanced/preferences.html).
+GeNN can simulate models written for the [Brian simulator](http://brian2.readthedocs.io) via the [Brian2GeNN](http://brian2genn.readthedocs.io) interface \cite  brian2genn2018.
+In order for Brian2GeNN to find GeNN, GeNN must either be in your path with the location of the CUDA libraries configured via the `CUDA_PATH` environment variables (as described in \ref Installation) or the `devices.genn.path` and `devices.genn.cuda_backend.cuda_path` [Brian preferences](http://brian2.readthedocs.io/en/stable/advanced/preferences.html) should be set.
```
Comment (tnowotny) [inline doxygen/06_Brian2GeNN.dox:5]: "variable"

File: doxygen/00_MainPage.dox (hunk 1 above)
Comment (neworderofjamie) [reply doxygen/00_MainPage.dox:21]: This was intentional as this renders to:
> Currently there are interfaces for SpineML and SpineCreator, a Python interface (PyGeNN) and a Brian interface via Brian2GeNN.

Comment (tnowotny) [reply doxygen/00_MainPage.dox:21]: hum ... "and SpineCreator" appearing magically ;-) ?
Ok, fine by me.

**Pull request merged**